src/main/java/com/snowflake/kafka/connector/SnowflakeSinkTask.java

```diff
@@ -171,11 +171,16 @@ public void start(final Map<String, String> parsedConfig) {
         Boolean.parseBoolean(parsedConfig.get(SnowflakeSinkConnectorConfig.REBALANCING));
 
     // default to snowpipe
+    // If it is snowpipe_streaming, set delivery guarantee to exactly once.
     IngestionMethodConfig ingestionType = IngestionMethodConfig.SNOWPIPE;
     if (parsedConfig.containsKey(SnowflakeSinkConnectorConfig.INGESTION_METHOD_OPT)) {
       ingestionType =
           IngestionMethodConfig.valueOf(
               parsedConfig.get(SnowflakeSinkConnectorConfig.INGESTION_METHOD_OPT).toUpperCase());
+      if (ingestionType.equals(IngestionMethodConfig.SNOWPIPE_STREAMING)) {
+        ingestionDeliveryGuarantee =
+            SnowflakeSinkConnectorConfig.IngestionDeliveryGuarantee.EXACTLY_ONCE;
+      }
```

Comment:
Should this be in the `validateConfig` file below? Might be easier to consolidate the logic that determines which ingestion delivery guarantee to use.

Reply:
Validation is already done in Utils.validateConfig.. here we just fetch the values from config so that we can decide which SnowflakeSinkService to initialize.